woocommerce/woocommerce-ios
# Errors: Banner message for error loading data on Orders and Reviews lists

Part of: #4382 
Relies on: #4400

## Description

To improve our error states, we are introducing a top banner that can be used whenever there is a problem loading data. The banner is collapsed by default and on select, users can read our troubleshooting tips or contact support directly.

This PR adds the banner to the Orders and Reviews lists. It also adds the banner to the Empty State screen, so it can be shown/hidden as needed when that screen is used.

## Changes

* Adds the error banner to `EmptyStateViewController`. The banner appears above the stack view containing the empty state content (title, image, message, and button).
* Updates `EmptyStateViewController.xib` to add the `stackView` outlet (for the banner's layout constraints). Other changes to that file were made automatically when the outlet was added.
* Adds the error banner to `OrderListViewController`. Any sync errors will set `hasErrorLoadingData` to `true`, which determines whether the error banner is displayed on the order list or the order empty state screen.
* Adds the error banner to `ReviewsViewController`. Any sync errors will set `hasErrorLoadingData` to `true`, which determines whether the error banner is displayed on the order list or the order empty state screen.

/|Collapsed (empty state)|Expanded (empty state)|Expanded (cached data)
-|-|-|-
Orders (Processing)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-11 at 22 01 24](https://user-images.githubusercontent.com/8658164/121750172-caa3b980-cb03-11eb-94b3-a122a686278d.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-11 at 22 01 51](https://user-images.githubusercontent.com/8658164/121750186-d2fbf480-cb03-11eb-91aa-c905f3bb9216.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-11 at 22 08 26](https://user-images.githubusercontent.com/8658164/121750197-d7281200-cb03-11eb-9962-417c1f3f5cae.png)
Orders (All Orders)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-11 at 22 02 50](https://user-images.githubusercontent.com/8658164/121750228-e4dd9780-cb03-11eb-984d-667f260be29b.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-11 at 22 03 02](https://user-images.githubusercontent.com/8658164/121750233-e60ec480-cb03-11eb-8341-c3cbdd854937.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-11 at 22 08 29](https://user-images.githubusercontent.com/8658164/121750238-e7d88800-cb03-11eb-8ab4-72a433574f78.png)
Reviews|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-11 at 22 01 34](https://user-images.githubusercontent.com/8658164/121750258-f3c44a00-cb03-11eb-8b06-40b233ca0c74.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-11 at 22 01 39](https://user-images.githubusercontent.com/8658164/121750262-f4f57700-cb03-11eb-99fd-bb0315f7ae03.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-06-11 at 22 26 31](https://user-images.githubusercontent.com/8658164/121750388-34bc5e80-cb04-11eb-8d1a-d91bd7e7b390.png)



## Testing

1. Build and run a fresh install of the app (to begin without any cached data).
2. Log in to the app.
3. Disconnect your internet connection. (This is so you will trigger a sync error; you can also disrupt your store's Jetpack connection for a more realistic scenario.)
4. Select Orders to open the product list.
5. Notice the error banner appears at the top of the Orders empty state screen on both the "Processing" and "All Orders" tabs.
6. Expand the banner and confirm the "Troubleshoot" button opens a troubleshooting doc, and "Contact support" opens a new support ticket in the app.
7. Reconnect to the internet.
8. Pull to refresh the Orders list. When your orders load, notice the error banner disappears.
9. Disconnect from the internet again, pull to refresh, and notice the error banner appears again above the cached orders.
10. Repeat steps 3-9 on the Reviews tab.

## Submitter Checklist

Update release notes:
